.kiro/steering/aws-sdk-java-v2-general.md

```diff
@@ -0,0 +1,162 @@
+
+- If the class's primary purpose is to return instances of another class:
+  - If the "get" method has no parameters:
+    - If the class implements `Supplier`: `{Noun}Supplier` (e.g. `CachedSupplier`)
+    - If the class does not implement `Supplier`: `{Noun}Provider` (e.g. `AwsCredentialsProvider`)
+  - If the "get" method has parameters: `{Noun}Factory` (e.g. `AwsJsonProtocolFactory`)
+
+#### Service-specific classes
+
+- If the class makes service calls:
+  - If the class can be used to invoke *every* data-plane operation:
+    - If the class is code generated:
+      - If the class uses sync HTTP: `{ServiceName}Client` (e.g. `DynamoDbClient`)
+      - If the class uses async HTTP: `{ServiceName}AsyncClient` (e.g. `DynamoDbAsyncClient`)
+    - If the class is hand-written:
+      - If the class uses sync HTTP: `{ServiceName}EnhancedClient` (e.g. `DynamoDbEnhancedClient`)
+      - If the class uses async HTTP: `{ServiceName}EnhancedAsyncClient` (e.g. `DynamoDbEnhancedAsyncClient`)
+  - If the class can be used to invoke only *some* data-plane operations:
+    - If the class uses sync HTTP: `{ServiceName}{Noun}Manager` (e.g. `SqsBatchManager`)
+    - If the class uses async HTTP: `{ServiceName}Async{Noun}Manager` (e.g. `SqsAsyncBatchManager`)
+    - Note: If only one implementation exists and it uses async HTTP, `Async` may be excluded. (e.g. `S3TransferManager`)
+- If the class does not make service calls:
+  - If the class creates presigned URLs: `{ServiceName}Presigner` (e.g. `S3Presigner`)
+  - If the class is a collection of various unrelated "helper" methods: `{ServiceName}Utilities` (e.g. `S3Utilities`)
+
+## Class Initialization
+
+- Favor static factory methods over constructors:
+  - Static factory methods provide meaningful names compared with constructors
+  - They are useful when working with immutable classes as we can reuse the same object
+  - Static factory methods can return any subtype of that class
+
+### Naming Conventions for Static Factory Methods
+- `create()`, `create(params)` when creating a new instance (e.g., `DynamoDBClient.create()`)
+- `defaultXXX()` when returning an instance with default settings (e.g., `BackoffStrategy.defaultStrategy()`)
+
+## Use of Optional
+
+- `Optional` **MUST NOT** be used when the result will never be null
+- For return types:
+  - `Optional` **SHOULD** be used when it is not obvious to a caller whether a result will be null
+  - `Optional` **MUST NOT** be used for "getters" in generated service model classes
+- For member variables: `Optional` **SHOULD NOT** be used
+- For method parameters: `Optional` **MUST NOT** be used
+
+## Object Methods (toString, equals, hashCode)
+
+- All public POJO classes **MUST** implement `toString()`, `equals()`, and `hashCode()` methods
+- When adding new fields to existing POJO classes, these methods **MUST** be updated to include the new fields
+- Implementation guidelines:
+  - `toString()`: Include class name and all fields with their values
```

Comment:
Possible additional guideline - do not include fields that could be considered sensitive such as credentials